lib/logstash/devutils/rspec/shared_examples.rb

```diff
@@ -0,0 +1,23 @@
+require 'rspec/wait'
+
+RSpec.shared_examples "an interruptible input plugin" do
+  describe "#stop" do
+    let(:queue) { SizedQueue.new(20) }
+    subject do
+      s = described_class.new(config)
+      s.register
+      s
+    end
+    it "returns from run" do
+      consumer_thread = Thread.new(queue) { |queue| loop { queue.pop } }
+      plugin_thread = Thread.new(subject, queue) { |subject, queue| subject.run(queue) }
+      # the run method is a long lived one, so it should still be running after "a bit"
+      sleep 0.5
```

Comment:
I agree with @guyboertje in here, I don't think stuff like the subscribe should be an issue implementing this kind of behaviour. If using callbacks like happen here we more or less know always where to send the ready notification. 

As I shared in https://github.com/elastic/logstash/pull/3812#issuecomment-137469431 I do really think this will make stuff easier. 

What do you think?